numpy/core/memmap.py

```diff
@@ -335,3 +341,48 @@ def __getitem__(self, index):
         if type(res) is memmap and res._mmap is None:
             return res.view(type=ndarray)
         return res
+
+    def close(self):
+        """
+        Close the memmap file and reduce reference count by 1.
+        If the reference count is zero, the file is unlocked.
+        
+        It has no effect if no file is memory-mapped.
+        For further information, see `memmap`.
+        
+        Parameters
+        ----------
+        None
+
+        See Also
+        --------
+        memmap
+        
+        """
+        if self._mmap is not None:
+            self._mmap.close()
+            self._mmap = None
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.close()
+
+    def closed(self):
+        """
+        Return True if the memmap is already closed, 
+        otherwise return False.
+
+        For further information, see `memmap`.
+
+        Parameters
+        ----------
+        None
+
+        See Also
+        --------
+        memmap
+
+        """
+        return True if self._mmap is None else False
```

Comment:
```suggestion
        return True if self._mmap is None else self._mmap.closed
```
If `self._mmap is not None` it does not necessarily mean that the `mmap.mmap` is still opened.

Reply:
You are right. Thanks.